### Python Review Assignment
16 points

Due by midnight February 1, 2024


**To avoid erasing the sample output, please insert a new cell above the sample output**. Write your script inside the new cell and run. Verify your output with the sample output

For lab submission, please rename your file using the following format

`firstname_familyname_python_review.ipynb`

In [ ]:
# please enter your name and 700# in this cell




### Task 1
Define a string `s1` with content 'Good morning John'. Count the number of characters in `s1`

In [ ]:
s1 = 'Good morning John'


17

### Task 2
Check if the word 'morning' is contained in `s1`

True

### Task 3
Use a function to replace 'John' with 'David' in `s1`. Show the content of the new string

'Good morning David'

### Task 4
Use your UCM email address to create a string s2. Extract the string `ucmo.edu`

'ucmo.edu'

### Task 5
Create a list x with integers from 1 to 30. Create a new list y which only conatins the odd numbers from list x. You are not allowed to mannually type out the lists

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29]

### Task 6
Create a new list z with each element is a squre of the corresponding element from list y. You are not allowed to mannually type out the list

[1, 9, 25, 49, 81, 121, 169, 225, 289, 361, 441, 529, 625, 729, 841]

### Task 7
Extract `[729, 625, 529, 441, 361, 289, 225, 169]` from list z. You are not allowed to mannually type out the list

[729, 625, 529, 441, 361, 289, 225, 169]

### Task 8
Reverse the order of z. You are not allowed to mannually type out the list

[841, 729, 625, 529, 441, 361, 289, 225, 169, 121, 81, 49, 25, 9, 1]

### Task 9
Given the following list, extract the string 'Get Me'

In [ ]:
lst = [1, 2, [3, 4], [5, [100, 200, ['Get Me']], 300, 11], 1, 9]


'Get Me'

### Task 10
Sort the following list of strings by their lengths. You are not allowed to mannually type out the list

In [ ]:
x = ['ccc', 'dddd', 'bb', 'ggggggg', 'a', 'eeeee', 'ffffff']


['a', 'bb', 'ccc', 'dddd', 'eeeee', 'ffffff', 'ggggggg']

### Task 11
Extract the string 'Get Me' from the following dictionary d

In [ ]:
d = {'k1':[1, 2, 3, {'tricky':['oh', 'man', 'crazy', {'target':[1, 2, 3, 'Get Me']}]}]}


'Get Me'

### Task 12
Use lambda expressions and the `filter()` function to filter out words from the following list that don't start with the letter 's'

In [ ]:
seq = ['soup', 'apple', 'salad', 'banana', 'spaghetti', 'spinach', 'kiwi']


['apple', 'banana', 'kiwi']

### Task 13
Count the number of unique characters in the following string s. Please ignore the white space and character case. For example, 'P' and 'p' are considered as one character

In [ ]:
s = 'Python programming is fun'


14

### Task 14
Use lambda expressions and the `map()` function to count the length of each string in the list

In [1]:
seq = ['soup', 'apple', 'salad', 'banana', 'spaghetti', 'spinach', 'kiwi']


[4, 5, 5, 6, 9, 7, 4]

### Task 15
Create a function that counts the number of times the word 'dog' occurs in a string. Ignore edge cases. 

In [ ]:
countDog('This dog looks differently from that dog in the dog shelter!')


3

### Task 16
Write a `for` loop to process employee information listed in the cell below. For each employee with Level 1, increase the salary by 2000 and adjust the Level from 1 to 2. Display all employee information afterwards.

In [1]:
employee = {
    'Elon Musk':{'Department':'Sales', 'Salary':6000, 'Level':2},
    'Bill Gates': {'Department':'Engineering', 'Salary':10000, 'Level':3},
    'Donald Trump':{'Department':'Public Relations', 'Salary':3000, 'Level':1},
    'Joe Biden':{'Department':'Human Resources', 'Salary':5000, 'Level':2},
    'Mark Zuckerberg':{'Department':'Engineering', 'Salary':10000, 'Level':3},
    'Jensen Huang':{'Department':'Sales', 'Salary':4000, 'Level':1}
}

{'Elon Musk': {'Department': 'Sales', 'Salary': 6000, 'Level': 2}, 'Bill Gates': {'Department': 'Engineering', 'Salary': 10000, 'Level': 3}, 'Donald Trump': {'Department': 'Public Relations', 'Salary': 5000, 'Level': 2}, 'Joe Biden': {'Department': 'Human Resources', 'Salary': 5000, 'Level': 2}, 'Mark Zuckerberg': {'Department': 'Engineering', 'Salary': 10000, 'Level': 3}, 'Jensen Huang': {'Department': 'Sales', 'Salary': 6000, 'Level': 2}}
